src/ol_infrastructure/infrastructure/vector_log_proxy/__main__.py

```diff
@@ -0,0 +1,345 @@
+"""Create the resources needed to run a vector-log-proxy server.
+
+"""
+import base64
+import json
+import textwrap
+from pathlib import Path
+
+import pulumi_vault as vault
+import yaml
+from pulumi import Config, StackReference
+from pulumi_aws import acm, autoscaling, ec2, get_caller_identity, iam, lb, route53
+
+from bridge.secrets.sops import read_yaml_secrets
+from ol_infrastructure.lib.aws.ec2_helper import (
+    DiskTypes,
+    InstanceTypes,
+    default_egress_args,
+)
+from ol_infrastructure.lib.consul import get_consul_provider
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.vault import setup_vault_provider
+
+##################################
+##    Setup + Config Retrival   ##
+##################################
+
+if Config("vault_server").get("env_namespace"):
+    setup_vault_provider()
+stack_info = parse_stack()
+vector_log_proxy_config = Config("vector_log_proxy")
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+policy_stack = StackReference("infrastructure.aws.policies")
+dns_stack = StackReference("infrastructure.aws.dns")
+consul_stack = StackReference(f"infrastructure.consul.operations.{stack_info.name}")
+mitodl_zone_id = dns_stack.require_output("odl_zone_id")
+
+target_vpc_name = (
+    vector_log_proxy_config.get("target_vpc") or f"{stack_info.env_prefix}_vpc"
+)
+target_vpc = network_stack.require_output(target_vpc_name)
+
+VECTOR_LOG_PROXY_PORT = vector_log_proxy_config.get("listener_port") or 9000
+
+consul_security_groups = consul_stack.require_output("security_groups")
+aws_config = AWSBase(
+    tags={"OU": "operations", "Environment": f"operations-{stack_info.env_suffix}"}
+)
+aws_account = get_caller_identity()
+ops_vpc_id = target_vpc["id"]
```

Comment:
Since we're being agnostic to which VPC we should change this naming to just `vpc_id` or `target_vpc_id` or similar